src/control/positionControl/pathPlanning/VoronoiPathPlanning.cpp

```diff
@@ -0,0 +1,131 @@
+
+void VoronoiPathPlanning::generateGraphFromDiagram() {
+    graphAdjacencyList.clear();
+    for (const jcv_edge* edgeIterator = jcv_diagram_get_edges(&voronoiDiagram); edgeIterator != nullptr; edgeIterator = edgeIterator->next) {
+        // ignore edges of the diagram
+        if (edgeIterator->a == 0 && edgeIterator->b == 0 && edgeIterator->c == 0) {
+            continue;
+        }
+
+        double distance = Vector2(edgeIterator->pos[0].x - edgeIterator->pos[1].x, edgeIterator->pos[0].y - edgeIterator->pos[1].y).length();
+
+        // ignore degenerate lines
+        if (distance == 0) {
+            continue;
+        }
+
+        Vector2 firstEdgePoint = convertFromJcvPoint(edgeIterator->pos[0]);
+        Vector2 secondEdgePoint = convertFromJcvPoint(edgeIterator->pos[1]);
+
+        // if the vertex is a new one (not found in the map), add it
+        if (graphAdjacencyList.empty() || graphAdjacencyList.find(firstEdgePoint) == graphAdjacencyList.end()) {
+            graphAdjacencyList[firstEdgePoint] = std::list<GraphNode>(1, (GraphNode){secondEdgePoint, distance});
+        }
+        // else append the adjacency list with the new neighbouring site and the new node connection
+        else {
+            graphAdjacencyList[firstEdgePoint].push_back((GraphNode){secondEdgePoint, distance});
+        }
+
+        // repeat for the second node
+        if (graphAdjacencyList.empty() || graphAdjacencyList.find(secondEdgePoint) == graphAdjacencyList.end()) {
+            graphAdjacencyList[secondEdgePoint] = std::list<GraphNode>(1, (GraphNode){firstEdgePoint, distance});
+        } else {
+            graphAdjacencyList[secondEdgePoint].push_back((GraphNode){firstEdgePoint, distance});
+        }
+    }
+}
+
+void VoronoiPathPlanning::addPointToGraph(const Vector2& pointToAdd) {
+    // 1 or fewer obstacles in the area - no points in the graph
+    if (graphAdjacencyList.empty()) {
+        graphAdjacencyList[pointToAdd] = {};
+        return;
+    }
+
+    double minDist = -1;
+    Vector2 closestPoint;
+    // add point to the graph, connecting it to the closest node
+    for (auto const& positionIterator : graphAdjacencyList) {
+        if (minDist < 0 || (positionIterator.first - pointToAdd).length() < minDist) {
+            closestPoint = positionIterator.first;
+            minDist = (positionIterator.first - pointToAdd).length();
+        }
+    }
+    if (graphAdjacencyList.find(pointToAdd) == graphAdjacencyList.end()) {
+        graphAdjacencyList[pointToAdd] = std::list<GraphNode>(1, (GraphNode){closestPoint, minDist});
+        graphAdjacencyList[closestPoint].push_back({pointToAdd, minDist});
+    }
+}
+
+std::vector<Vector2> VoronoiPathPlanning::generatePathDijkstra(const Vector2& initialPosition, const Vector2& targetPosition) {
+    std::unordered_map<Vector2, float> distanceVector;
+    std::unordered_map<Vector2, Vector2> parentVector;
+    distanceVector.insert({initialPosition, 0});
+    std::list<Vector2> nodeQueue;
```

Comment:
nodeQueue, in an std::list?...

Reply:
The algorithm at its base uses a queue, but I had some problems implementing it as one, so I made it a list. The elements get traversed only once, so I wouldn't name it a list